src/qt/sendcoinsdialog.cpp

```diff
@@ -293,37 +290,29 @@ void SendCoinsDialog::on_sendButton_clicked()
         {
             // Unlock wallet was cancelled
             fNewRecipientAllowed = true;
-            return;
+            return false;
         }
-        send(recipients);
-        return;
     }
-    // already unlocked or not encrypted at all
-    send(recipients);
-}
 
-void SendCoinsDialog::send(QList<SendCoinsRecipient> recipients)
-{
```

Comment:
17509: this should be changed to 
```
        return send(recipients);
    }
    // already unlocked or not encrypted at all
    return send(recipients);
}

bool SendCoinsDialog::send(QList<SendCoinsRecipient> recipients)
{
```
, not dropped. `UnlockContext` is important.